### ISSUE TITLE
[Issue]: TE dropout requires old-style PRNGs

### ISSUE THREAD

**Issue by gabeweisz** [ROCm/TransformerEngine#244]
### Problem Description

Jax changed the preferred type for RNG seeds from an array of ints to an opaque type in [August 2023](https://docs.jax.dev/en/latest/jep/9263-typed-keys.html#overview).

TE attention still requires the old-style keys in [transformer_engine/jax/cpp_extensions/attention.py](https://github.com/ROCm/TransformerEngine/blob/c97ecae53e8cd1d8b925530eef25036ef43f9181/transformer_engine/jax/cpp_extensions/attention.py#L208).
Any applications using new-style keys will see an error when this line of code attempts to convert the seed to an int.

It may be possible to work around this issue easily by changing the line to:
`seed = jax.random.key_data(seed).astype(self.rng_state_dtype)`

### Operating System

Ubuntu 22.04, Ubuntu 24.04

### CPU

Any

### GPU

MI300x series

### ROCm Version

Any 6.4

### Steps to Reproduce

Run TE DotProductAttention with dropout != 0.0 (for example set it to 0.2) and a PRNG created with  jax.random.key() instead of jax.random.PRNGKey()

**Comment by gabeweisz** [comment]
Adding this two line update to line 208 fixes the crash:

if (jnp.issubdtype(seed.dtype, jax.dtypes.prng_key)):
  seed = jax.random.key_data(seed)
